bootstrap/src/test/java/io/airlift/bootstrap/TestLifeCycleManager.java

```diff
@@ -212,6 +215,76 @@ public void testJITInjection()
         assertEquals(stateLog, ImmutableList.of("postDependentInstance", "preDependentInstance"));
     }
 
+    @Test
+    public void testPreDestroySuppressedExceptionHandling()
+            throws Exception
+    {
+        Injector injector = Guice.createInjector(
+                Stage.PRODUCTION,
+                new LifeCycleModule(),
+                binder -> {
+                    binder.bind(DependentInstance.class).in(Scopes.SINGLETON);
+                    binder.bind(DestroyExceptionInstance.class).in(Scopes.SINGLETON);
+                });
+
+        LifeCycleManager lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        lifeCycleManager.start();
+        assertEquals(stateLog, ImmutableList.of("postDependentInstance"));
+        try {
+            lifeCycleManager.stopWithoutFailureLogging();
+            fail("Expected exception to be thrown");
+        }
+        catch (LifeCycleStopException e) {
+            assertEquals(e.getSuppressed().length, 2, "Expected two suppressed exceptions");
+            Set<String> suppressedMessages = Arrays.stream(e.getSuppressed())
+                    .map(Throwable::getMessage)
+                    .collect(Collectors.toSet());
+            assertEquals(ImmutableSet.copyOf(suppressedMessages), ImmutableSet.of("preDestroyExceptionOne", "preDestroyExceptionTwo"));
+        }
+        catch (Exception e) {
```

Comment:
Let's remove this catch block. Better to let the exception propagate and fail the test, since that's the normal behavior, and it will provide the full stack trace of the unexpected exception.